src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1142,5 +1142,21 @@ internal static bool IsAzureSqlServerEndpoint(string dataSource)
 
             return false;
         }
+
+        static internal ArgumentOutOfRangeException InvalidDataRowVersion(DataRowVersion value)
+        {
+#if DEBUG
```

Comment:
Would be worth verifying in the compiled IL, but I would expect that this #if DEBUG isn't necessary; in a release build, all of the cases will be empty (since the Debug.Assert will evaporate), and I would expect nothing would get emitted for the switch.  I could be wrong, though.